marlowe-dashboard-client/src/Contract/Types.purs

```diff
@@ -14,31 +14,41 @@ import Data.Map (Map)
 import Data.Maybe (Maybe(..))
 import Data.Set (Set)
 import Halogen (RefLabel(..))
-import Marlowe.Execution (ExecutionState, NamedAction)
+import Marlowe.Execution.Types (NamedAction)
+import Marlowe.Execution.Types (State) as Execution
 import Marlowe.Extended.Metadata (MetaData)
 import Marlowe.PAB (PlutusAppId, MarloweParams)
 import Marlowe.Semantics (ChoiceId, ChosenNum, Party, Slot, TransactionInput, Accounts)
 import WalletData.Types (WalletDetails, WalletNickname)
 
+-- The Contract.State wraps the Execution.State up with some additional information that is necessary
+-- to render the execution state of the contract in the app. Some of this is derived data that could
+-- be calculated on the fly, but is stored here for efficiency. Some of it is data specific to the UX
+-- (like which step card is currently centered, and which tabs are shown in which step cards). And some
+-- of it is metadata (the actual metadata of the contract, but also functional metadata like the ID of
+-- the user's follower app for this contract, and the MarloweParams of the contract).
```

Comment:
Not sure what is meant by this comment, maybe we could add the pertinent parts to the fields that need clarification, for example the fields that are stored for efficiency.

We need to keep in mind that this component changed scope since it's inception, so just `Contract` might not be the best name. Maybe it would help if we renamed this to ContractCarousel, and identify parts that we'll need to separate when we try to add the same/similar visualization to the simulator.

This is however a Component folder, so explaining 
```
Some of it is data specific to the UX (like which step card is currently centered, and which tabs are shown in which step cards)
```
is redundant.